hydra-node/test/Hydra/BehaviorSpec.hs

```diff
@@ -689,7 +690,7 @@ withHydraNode ::
 withHydraNode signingKey otherParties connectToChain action = do
   outputs <- atomically newTQueue
   outputHistory <- newTVarIO mempty
-  nodeState <- createNodeState $ IdleState{chainState = SimpleChainState{slot = ChainSlot 0}}
+  nodeState <- createNodeState $ Idle IdleState{chainState = SimpleChainState{slot = ChainSlot 0}}
```

Comment:
Suggestion: This feels awkward, provide a "smart constructor"?

Reply:
I'd rather not. The smart constructors would need to use positional arguments or yet another type that is isomorphic to `IdleState` et al.

Reply:
What arguments? There's none here

Reply:
My commment applies under the assumption you mean a function like `mkIdleState :: ChainStateType tx -> HeadState tx` as "smart constructor". So this would read `nodeState <- createNodeState $ mkIdleState SimpleChainState{slot = ChainSlot 0}`.

Maybe I misunderstood you and I would need more concrete advice to consider your suggestion :)